syntax = "proto3";


service VideoService {
  rpc GetVideo (VideoRequest) returns (VideoResponse) {}
  rpc GetCategories (CategoryRequest) returns (CategoryResponse) {}
}


message CategoryResponse {
    repeated Category categories = 1;
    string response_code = 2;
}


message CategoryRequest {
  string request = 1;
}


message VideoResponse {
    repeated Video videos = 1;
    string response_code = 2;
}


message VideoRequest {
  int32 user_id = 1;
  int32 video_id = 2;
  int32 category_id = 3;
  bool latest = 4;
  Video video = 5;
}

message Video {
  int32 user_id = 1;
  int32 video_id = 2;
  string title = 3;
  string resume = 4;
  int32 category_id = 5;  
}

message Category {
  string category = 1;
  int32 category_id = 2;  
}